pkg/envoy/rds/route/route_config.go

```diff
@@ -286,6 +287,40 @@ func buildWeightedCluster(weightedClusters mapset.Set) *xds_route.WeightedCluste
 	sort.Stable(clusterWeightByName(wc.Clusters))
 	return &wc
 }
+func buildRetryPolicy(retry *v1alpha1.RetryPolicySpec) *xds_route.RetryPolicy {
+	rp := &xds_route.RetryPolicy{}
+	if retry == nil {
+		return rp
+	}
+	if retry.RetryOn != "" {
+		rp.RetryOn = retry.RetryOn
+	} else {
+		log.Error().Msgf("Retry policy must specify which policies to retry on")
+		return rp
+	}
+	if retry.NumRetries != 0 {
+		rp.NumRetries = &wrapperspb.UInt32Value{
+			Value: uint32(retry.NumRetries),
+		}
+	}
+	if retry.PerTryTimeout != 0 {
+		rp.PerTryTimeout = convertIntToDuration(retry.PerTryTimeout)
+	}
+	if retry.RetryBackoffInterval != 0 {
+		rp.RetryBackOff = &xds_route.RetryPolicy_RetryBackOff{
+			BaseInterval: convertIntToDuration(retry.RetryBackoffInterval),
+		}
+	}
+	return rp
```

Comment:
I have seen that these fields are optional, please could you specify somewhere in the code what these defaults would be when not inputted

Reply:
added defaults as comment to pkg/apis/policy/v1alpha1/retry.go https://github.com/openservicemesh/osm/pull/4476/files#diff-a817187c2d023b478ee6a5d9c91503db2b9a30ee45d6fb90e12e999897ae8b21R63-R81

Reply:
@shalier please do not add policy defaults in the Go types based on Envoy defaults. Kindly make all the values mandatory if possible. If we start adding default values for the fields, we need to re-review the design. Could you share the feature doc where we can discuss this?

Reply:
https://docs.google.com/document/d/1BF3kC7RRY_rXTKZxwxQMNiIzKoa3FVnX/edit?usp=sharing&ouid=115555270846252379013&rtpof=true&sd=true

Reply:
Will have retry policy fields all be required for now and will adjust as needed in a future PR